tests/agent_server/test_bash_service.py

```diff
@@ -0,0 +1,395 @@
+    stdout_outputs = [o for o in collector.outputs if o.stdout]
+    stderr_outputs = [o for o in collector.outputs if o.stderr]
+
+    # Should have both stdout and stderr
+    assert len(stdout_outputs) >= 1
+    assert len(stderr_outputs) >= 1
+
+    # Verify content
+    combined_stdout = "".join(o.stdout or "" for o in stdout_outputs)
+    combined_stderr = "".join(o.stderr or "" for o in stderr_outputs)
+
+    assert "stdout message" in combined_stdout
+    assert "stderr message" in combined_stderr
+
+    # All outputs should have exit code 0
+    for output in collector.outputs:
+        assert output.exit_code == 0
+
+
+@pytest.mark.asyncio
+async def test_command_with_error_exit_code(bash_service):
+    """Test bash command that exits with error code."""
+    collector = EventCollector()
+    await bash_service.subscribe_to_events(collector)
+
+    # Command that exits with error
+    command = BashCommand(command="exit 42", cwd="/tmp")
+    await bash_service.start_bash_command(command)
+
+    # Wait for command to complete
+    await asyncio.sleep(1)
+
+    # Verify events were published
+    assert len(collector.commands) == 1
+    assert len(collector.outputs) >= 1
+
+    # Verify exit code is propagated
+    for output in collector.outputs:
+        assert output.exit_code == 42
+
+
+@pytest.mark.asyncio
+async def test_command_timeout(bash_service):
+    """Test bash command that times out."""
+    collector = EventCollector()
+    await bash_service.subscribe_to_events(collector)
+
+    # Command that should timeout (sleep longer than timeout)
+    command = BashCommand(command="sleep 10", cwd="/tmp", timeout=1)
+    await bash_service.start_bash_command(command)
+
+    # Wait for timeout to occur plus some buffer
+    await asyncio.sleep(3)
+
+    # Verify events were published
+    assert len(collector.commands) == 1
+
+    # The timeout might not generate output events in all cases
+    # Just verify the command was started
+    cmd_event = collector.commands[0]
+    assert cmd_event.command == "sleep 10"
+
+
+@pytest.mark.asyncio
+async def test_large_output_chunking(bash_service):
+    """Test that large output is properly chunked."""
+    collector = EventCollector()
+    await bash_service.subscribe_to_events(collector)
+
+    # Generate large output using a simple command that should work everywhere
+    # Create a string larger than MAX_CONTENT_CHAR_LENGTH (1MB)
+    large_size = 1024 * 1024 + 1000  # Slightly over 1MB
+    command = BashCommand(command=f'yes "x" | head -c {large_size}', cwd="/tmp")
+    await bash_service.start_bash_command(command)
+
+    # Wait for command to complete
+    await asyncio.sleep(5)
+
+    # Verify events were published
+    assert len(collector.commands) == 1
+    assert len(collector.outputs) >= 1  # Should be chunked if large enough
+
+    # Verify all chunks belong to same command and are ordered
+    for i, output in enumerate(collector.outputs):
```

Comment:
We should assert that this is long enough that contains more than 1 chunk if it is actually the case